experiments/compositions/composition/internal/controller/expander_reconciler.go

```diff
@@ -181,26 +179,31 @@ func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		// ------------------- EXPANSION SECTION -----------------------
 		logger = loggerCR.WithName(expander.Name).WithName("Expand")
 
-		expanderVersion, expanderRegistry, err := r.getExpanderVersion(ctx, expander.Version, expander.Type)
+		version := expander.Version
+		if version == "" {
+			version = "latest"
+		}
+		value, ev, err := r.getExpanderVersion(ctx, version, expander.Type)
 		if err != nil {
 			reason := "FailedGettingValidExpanderVersion"
 			if apierrors.IsNotFound(err) {
 				// The CR should be created before the specified expander can be used.
 				reason = "MissingExpanderCR"
-				logger.Error(err, "Failed to get the ExpanderVersionCR", "expander", expander.Type, "reason", reason)
+				logger.Error(err, "Failed to get the ExpanderVersionCR")
```

Comment:
Since each expander (eg jinja2 and others, etc) will have a ExpanderVersionCR. If the CR is missing, why wouldn't we want to show in the log that the type and the expander that is missing?

Reply:
The logger has the expander.Name as part of it.
But the `expander.type` is definitely a good idea.

`158:			logger = loggerCR.WithName(expander.Name).WithName("Fetcher")`

Reply:
we record this info in the line below:

```
197:			logger.Error(err, "Error getting expander version", "expander", expander.Type,
198:				"version", version, "reason", reason)
```

Reply:
refactored.